### PR TITLE
feat(model): Refactoring AI Tasks for Consistency Across Text and Image Generation

### PR DESCRIPTION
**Pull request by tonywang10101**
This commit introduces a refactoring of several AI tasks to ensure consistency and clarity across our text and image generation functionalities. Our main focus is to address the inconsistencies previously observed in how certain tasks were categorized.

Changes include:

- Separating 'text generate chat', 'text generate', and 'visual question answering' under the umbrella of text generation tasks.
- Reorganizing 'text2image' and 'image2image' tasks into a more coherent image generation category.

This restructuring is in alignment with the Linear ticket INS-2574, ensuring a more streamlined and intuitive categorization of AI tasks. This update makes it easier for users and developers to navigate and understand our AI task offerings, enhancing the overall user experience.

### CODE REVIEW

**Comment by heiruwu** [comment]
also you might want to take a look at the CodeQL scanning result, not sure if it will be a problem

File: pkg/db/migration/000001_init.up.sql [hunk 1]
```diff
@@ -20,7 +20,10 @@ CREATE TYPE valid_task AS ENUM (
     'TASK_INSTANCE_SEGMENTATION',
     'TASK_SEMANTIC_SEGMENTATION',
     'TASK_TEXT_TO_IMAGE',
-    'TASK_TEXT_GENERATION'
+    'TASK_IMAGE_TO_IMAGE',
```
[the remark this answers is not in the document]
Comment (heiruwu) [reply pkg/db/migration/000001_init.up.sql:23]: I will deal with this along with another potential bug in the migration script in the later PR, mark it as resolved for now

[the remark this answers is not in the document]
Comment (tonywang10101) [reply pkg/db/migration/000001_init.up.sql:23]: Got it, thanks!

**Review by heiruwu — APPROVED** [review]
LGTM

**Pull request merged**